src/user/service.ts

```diff
@@ -117,3 +118,65 @@ export async function validateUserCredentials(user: User, password: string, log:
   return true;
 }
 
+/**
+ * @description - Given a principal User, uses the principal id to find the UserInfo record for a user.
+ * @param user - The principal User to find the UserInfo record for.
+ * @returns The UserInfo record for the user.
+ * @throws NotFound - If the UserInfo record is not found.
+ */
+export async function findUserInfoByUser(user: User): Promise<UserInfo> {
+
+  const result = await db('user_info')
+    .select()
+    .where({principal_id: user.id})
+    .first();
+
+  if (!result) throw new NotFound(`UserInfo for user "${user.id}" not found.`);
+
+  return recordToModel(user, result);
+}
+
+/**
+ * @description - Given a principal User, uses the principal id to find the Principal user and update the UserInfo record for a user.
+ * @param user - The principal User to update the UserInfo record for.
+ * @param userInfo - new UserInfo object to update the UserInfo record with.
+ * @returns The updated UserInfo record.
+ * @throws BadRequest - If the UserInfo record is not updated.
+ */
+export async function updateUserInfo(user: User, userInfo?: UserInfo): Promise<void> {
+
+  if (!userInfo) return;
+
+  const result = await db('user_info')
+    .where({principal_id: user.id})
+    .update(
+      {
+        name: userInfo.name,
+        locale: userInfo.locale,
+        given_name: userInfo.givenName,
+        family_name: userInfo.familyName,
+        birthdate: userInfo.birthDate,
+        address: userInfo.address ? JSON.stringify(userInfo.address) : null,
+        zoneinfo: userInfo.zoneInfo,
```

Comment:
This should also set `updatedAt` to `Date.now()`.

Reply:
I did a global search and didn't see a precedence of `updatedAt` used in this codebase... is your intent is to have a non-namespace-collision use of `modifiedAt` on this `userInfo` type ?

Reply:
whoops I meant `modifiedAt` sorry!